src/map.rs

```diff
@@ -446,15 +527,59 @@ where
     }
 
     #[inline]
-    /// Maps `key` to `value` in this table.
+    /// Inserts a key-value pair into the map.
+    ///
+    /// If the map did not have this key present, [`None`] is returned.
+    ///
+    /// If the map did have this key present, the value is updated, and the old
+    /// value is returned. The key is not updated, though; this matters for
+    /// types that can be `==` without being identical. See the [std-collections
+    /// documentation] for more.
+    ///
+    /// [`None`]: std::option::Option::None
+    /// [std-collections documentation]: https://doc.rust-lang.org/std/collections/index.html#insert-and-complex-keys
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use flurry::HashMap;
+    ///
+    /// let map = HashMap::new();
+    /// assert_eq!(map.pin().insert(37, "a"), None);
+    /// assert_eq!(map.pin().is_empty(), false);
+    ///
+    /// map.pin().insert(37, "b");
+    /// assert_eq!(map.pin().insert(37, "c"), Some(&"b"));
+    /// assert_eq!(map.pin().get(&37), Some(&"c"));
+    /// ```
     ///
-    /// The value can be retrieved by calling [`HashMap::get`] with a key that is equal to the original key.
+    /// # Notes
```

Comment:
I wonder if we should move this note up to the type-level docs rather than having it hidden away on only some methods. What do you think?

Reply:
I agree, the notion of pinning transpires through the whole API, so might as well put in a place that will be more prominent in the docs.
Should I also remove the notes on the `iter()`, `keys()` and `values()` methods ?

Reply:
Yes, I think we should have it _just_ on `HashMap`.